lib/Models/RegionProvider.js

```diff
@@ -0,0 +1,405 @@
+    /**
+     * String uniquely identifying this type of region (eg, 'sa4')
+     * @type {String}
+     */
+    this.regionType = regionType;
+
+    /**
+     * Reference to the region provider list, to allow looking up disambiguating regions attributes.
+     * @type {Object}
+     */
+    this.regionProviderList = regionProviderList;
+    /** 
+    * WMS attribute whose value will correspond to each region's code.
+    * @type {String} 
+    */
+    this.regionProp = properties.regionProp;
+    /**
+    * A text description of this region type, which may feature in the user interface.
+    * @type {String}
+    */
+    this.description = properties.description;
+
+    /**
+    * Name of the WMS layer where these regions are found.
+    * @type {String}
+    */
+    this.layerName = properties.layerName;
+    /**
+    * URL of the WMS server
+    * @type {String}
+    */
+    this.server = properties.server;
+    /**
+     * List of aliases which will be matched against if found as column headings.
+     * @type {String[]}
+     */
+    this.aliases = defaultValue(properties.aliases, [this.regiontype]);
+    this.serverReplacements = properties.serverReplacements;
+    this.dataReplacements = properties.dataReplacements;
+    this.disambigColumns = properties.disambigColumns;
+    /** 
+     * Whether this region type uses text codes, rather than numeric. It matters because numeric codes are treated differently by the
+     * CSV handling models.
+     * @type {Boolean}
+     */
+    this.textCode = defaultValue(properties.textCodes, false); // yes, it's singular...
+
+    /**
+     * Array of attributes of each region, once retrieved from the server.
+     * @type {Object[]}
+     */
+    this.regions = [];
+
+
+};
+
+/**
+ * Turn a WFS list of attribute values into a straight array
+ * @param {String} propertyName: optional, for setting an additional (disambiguating) property on a region
+ *  Not sure if this is really necessary - we can surely access the name of that column when needed.
+ *
+ */
+RegionProvider.prototype.loadRegionsFromXML = function(xml, propertyName) {
+    var obj = xml2json(xml);
+
+    if (!defined(obj.member)) {
+        console.log(xml);
+        throw new ModelError("Couldn't load region boundaries for region " + this.regionProp);
+    }
+    
+    if (!(obj.member instanceof Array)) {
+        obj.member = [obj.member];
+    }
+    if (obj.member.length === 0) {
+        throw new ModelError("Zero region boundaries found for region " + this.regionProp);
+    }
+    for (var i = 0; i < obj.member.length; i++) {
+        if (this.regions[i] === undefined) {
+            this.regions[i] = {};
+        }
+        // either { id: 123 } or { id: 'Bunbury', STE_NAME: 'VIC' }
+        if (!propertyName) {
+            this.regions[i].id = obj.member[i][this.regionProp];
+        } else {
+            this.regions[i][propertyName] = obj.member[i][propertyName];
```

Comment:
Not sure if I'm following this right, but isn't it necessary to set the value of the `id` property even when a `propertyName` is supplied?